MonadicBitsTests/EitherTests.cs

```diff
@@ -60,8 +63,9 @@ public static void Match_left_either_returns_left_value()
         public static void Map_right_either_returns_either_with_new_type_and_value()
         {
             const int mappedValue = 42;
-            TestMonads.Right("Test").Map(_ => mappedValue)
-                .Match(Assert.Fail, i => Assert.AreEqual(mappedValue, i));
+            TestMonads.Right("Test")
+                .Map(_ => mappedValue)
+                .Should().Be(mappedValue.Right());
```

Comment:
This looks super good now. Really great for writing tests now.

Reply:
I think that will be the main usage. I am not 100% sure about the hash code implementations.